### DISCUSSION TITLE
Misleading description about combining common parts of oneOf in Factoring Schemas

### DISCUSSION THREAD

**Discussion by gh-andre** [json-schema-org/community#459] [Q&A]
Wonder if anyone can provide any insights into what the JSON Schema guide is saying about `oneOf` and how common parts can be "factored out" from `oneOf` parts. Here's the link and the text:

https://json-schema.org/understanding-json-schema/reference/combining.html?highlight=oneof#factoring-schemas

> Note that it’s possible to “factor” out the common parts of the subschemas. The following two schemas are equivalent.

This is the example.

```json
{
  "oneOf": [
    { "type": "number", "multipleOf": 5 },
    { "type": "number", "multipleOf": 3 }
  ]
}
```
```json
{
   "type": "number",
   "oneOf": [
     { "multipleOf": 5 },
     { "multipleOf": 3 }
   ]
 }
```
However, it appears that the text incorrectly characterizes that both schemas are equivalent, while the top one has 2 schemas and the bottom one has 3 schemas, each applied independently and not combined, as the text implies. 

If schemas were indeed combined as described, then it would be possible to introduce alternative fields, like this:
```json
{
  "additionalProperties": false,
  "properties": {  
    "a": {"type": "string"},
    "b": {"type": "boolean"}
  },
  "oneOf": [
    {"properties": {"x": {"type": "number"}}},
    {"properties": {"y": {"type": "number"}}}
  ]
}
```
However, because each schema is applied independently a document like this would fail, because `y` is not defined in the top schema and because both schemas in `oneOf` match, since each allows additional properties, meaning that outer `additionalProperties` is not applied here either:
```json
{
  "a": "abc", "b": true, "y": 123
}
```
Shouldn't the _Factoring Schemas_ instead say that in this specific case, the top schema and one of the `oneOf` branches are applied independently and for some configurations will appear like common parts are "factored out", but won't work in general?

**Comment by jdesrosiers** [comment]
> each applied independently and not combined, as the text implies

I'm not sure what you're getting at. I'm not seeing how the text implies that anything is combined. You are, of course, correct that there is no "combining" in this or any other aspect of JSON Schema.

**Comment by gh-andre** [comment]
@jdesrosiers 
> I'm not seeing how the text implies that anything is combined.

The text states that common text can be moved out of each schema ("factored out") and that both schema examples are equivalent, which to me suggests that schemas can be optimized in not to have to repeat the moved out parts, which is not the case.

**Reply by jdesrosiers** [reply]
"Equivalent" means that you will get the same true/false result from both schemas no matter what JSON instance you validate. That is true of these two schemas. The only time you can't refactor schemas like this is with keywords that depend on other keywords like how `additionalProperties` depends on `properties`. If you move one without the other, it won't work, but otherwise it's not a problem.

**Comment by gh-andre** [comment]
That's not how I read "equivalent" - the same outcome happens only in a particular use case, so this technique is not transferrable to other use cases. I will leave it at that, however. Thanks for your insights.
